src/finch/tensor.py

```diff
@@ -407,6 +419,35 @@ def construct_csf_jl_object(
     def construct_csf(cls, arg: TupleOf3Arrays, shape: tuple[int, ...]) -> "Tensor":
         return Tensor(cls.construct_csf_jl_object(arg, shape))
 
+    def to_scipy_sparse(self):
+        import scipy.sparse as sp
+
+        if self.ndim != 2:
+            raise ValueError("Can only convert a 2-dimensional array to a Scipy sparse matrix.")
+        if self.fill_value != 0:
+            raise ValueError("Can only convert arrays with 0 fill value to a Scipy sparse matrix.")
+        order = self.get_order()
+        body = self._obj.body
+
+        if str(jl.typeof(body.lvl).name.name) == "SparseCOOLevel":
+            data = np.asarray(body.lvl.lvl.val)
+            coords = body.lvl.tbl
+            row, col = coords[::-1] if order == (1, 0) else coords
+            row, col = np.asarray(row) - 1, np.asarray(col) - 1
+            return sp.coo_matrix((data, (row, col)), shape=self.shape)
+
+        if (
+            str(jl.typeof(body.lvl).name.name) == "DenseLevel" and
+            str(jl.typeof(body.lvl.lvl).name.name) == "SparseListLevel"
+        ):
+            data = np.asarray(body.lvl.lvl.lvl.val)
+            indices = np.asarray(body.lvl.lvl.idx) - 1
+            indptr = np.asarray(body.lvl.lvl.ptr) - 1
+            sp_class = sp.csr_matrix if order == (1, 0) else sp.csc_matrix
+            return sp_class((data, indices, indptr), shape=self.shape)
+
+        raise ValueError("Invalid format. Tensor should be a COO, CSR or CSC.")
```

Comment:
Let's convert to one of these formats first if we aren't already. If all the levels are dense ish we can convert to dense, otherwise COO is fine. Does Scipy do DCSR?

Reply:
> Let's convert to one of these formats first if we aren't already.

Sure! Let's convert to `COO` then as a default format.

> If all the levels are dense ish we can convert to dense

Dense format won't work, maybe if we have a dense tensor then I can throw an error?

In Scipy sparse, dense input isn't allowed:
```python
import scipy.sparse as sps
sps.linalg.inv(np.ones((3,3)))  # TypeError: Input must be a sparse matrix
```

> Does Scipy do DCSR?

Unfortunately not: https://docs.scipy.org/doc/scipy/reference/sparse.html#sparse-matrix-classes

Reply:
> Dense format won't work, maybe if we have a dense tensor then I can throw an error?

If we get a `np.ndarray` (dense), we can simply use `from_numpy` instead.

Reply:
I don't think I understand.

`def to_scipy_sparse` is meant to convert a Tensor to SciPy object only. We support `COO`, `CSR`, and `CSC` SciPy formats. In my opinion `to_scipy_sparse` should only return SciPy objects and never return a NumPy array. If Tensor is a dense one, a user shouldn't be able to call `to_scipy_sparse` on it.
I don't think `from_numpy` should be used here.

Reply:
Done!